Generate view updates in smaller parts

psarna opened the pull request:
In order to avoid large allocations and too large mutations
generated from large view updates, granularity of the process
is broken down from per-partition to smaller chunks.
The view update builder now produces partial updates, no more
than 100 view rows at a time.

The series was tested manually with a particular scenario in mind -
deleting a large base partition, which results in creating
a view update per each deleted row - which, with sufficiently
large partitions, can reach millions. Before the series, Scylla
experienced an out-of-memory condition after the view update
generation mechanism tried to load too much data into a contiguous
buffer. Multiple large allocation warnings and reactor stalls were
observed as well. After the series, the operation is still rather slow,
but does not induce reactor stalls nor allocator problems.
A reduced version of the above test is added as a unit test -
it does not check for huge partitions, but instead uses a number
just large enough to cause the update generation process to be split
into multiple chunks.

Fixes #8852

psarna commented:
I ran a scenario with 10M rows in a single base partition - removing it succeeded (and all 10M corresponding rows in the index were removed) and there were no stalls or large allocations in the system. The test was only possible after https://github.com/scylladb/scylla/pull/8876 was merged, because deleting such a large partition without custom timeout was not possible in my local environment.

psarna commented:
ping

avikivity commented on table.cc:
```diff
@@ -1666,17 +1666,20 @@ future<> table::generate_and_propagate_view_updates(const schema_ptr& base,
+        co_await db::view::mutate_MV(std::move(base_token), std::move(updates), _view_stats, *_config.cf_stats, std::move(tr_state),
+            std::move(units), service::allow_hints::yes, db::view::wait_for_all_updates::no);
+    } catch (...) {
+        // ignore
```
I see you preserved ignoring exceptions without explaining why. Please follow up with an explanation about why it is correct to ignore errors here.

avikivity commented on test/cql-pytest/test_secondary_index.py:
```diff
@@ -185,3 +185,18 @@ def test_paging_with_desc_clustering_order(cql, test_keyspace):
+            cql.execute(prep, [i])
+        cql.execute(f"DELETE FROM {table} WHERE p = 1")
+        res = [row for row in cql.execute(f"SELECT * FROM {table}_c1_idx_index")]
+        assert len(res) == 0
```
Please also add an example with a range tombstone (and verify that rows that should remain, remain).

psarna commented:
v2:
 - added a comment on why we ignore exceptions when propagating view updates for a table
 - added a range tombstone test case
Tests: unit(release)
